unison-core/src/Unison/Name.hs

```diff
@@ -103,20 +118,26 @@ stripNamePrefix prefix@Name{} name@Name{} =
   Name <$> Text.stripPrefix (toText prefix <> mid) (toText name)
   where
   mid = if toText prefix == "." then "" else "."
-stripNamePrefix (Name' prefix) (Name' name) =
-  fmap Name' do
-    suffix <- List.stripPrefix (toList prefix) (toList name)
-    NonEmpty.nonEmpty suffix
+stripNamePrefix n1@(Name' p1 prefix) n2@(Name' p2 name) =
+  if p1 == p2 then
+    fmap (Name' Relative) do
+      suffix <- List.stripPrefix (toList prefix) (toList name)
+      NonEmpty.nonEmpty suffix
+  else
+    error ("stripNamePrefix (" ++ show n1 ++ ") (" ++ show n2 ++ ")")
```

Comment:
I don't really know when this one is used.

Reply:
Same! I plan to audit all of the functions that currently call `error` and get rid of them long term... but for now this just rules out

```
stripNamePrefix .foo foo.bar
stripNamePrefix foo .foo.bar
```
which both seem sketchy (since they ignore one arg or the other's absolute/relative quality)